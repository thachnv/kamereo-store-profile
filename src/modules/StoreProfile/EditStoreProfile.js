import React, { useEffect, useState } from "react";
import * as yup from "yup";

import "antd/lib/icon/style/css";

import defaultAvatar from "../../assets/default-avatar.png";
import {
  Button,
  Divider,
  Form,
  Icon,
  Input,
  notification,
  Select,
  Spin,
  Upload
} from "antd";
import { ErrorMessage, Formik } from "formik";

const initialFormValues = {
  storeName: "",
  storeAddress: "",
  storeDistrict: "8",
  storeCity: "",
  storePhone: "",
  riCompanyName: "",
  riCompanyAddress: "",
  riCompanyDistrict: "8",
  riCompanyCity: "",
  riCompanyTaxCode: ""
};

export const formValuesToStoreProfile = values => {
  return {
    name: values.storeName,
    address: values.storeAddress,
    district: values.storeDistrict,
    city: values.storeCity,
    phone: values.storePhone,
    redInvoice: {
      name: values.riCompanyName,
      address: values.riCompanyAddress,
      district: values.riCompanyDistrict,
      city: values.riCompanyCity,
      taxCode: values.riCompanyTaxCode
    }
  };
};

export const storeProfileToFormValues = storeProfile => {
  return {
    storeName: storeProfile.name,
    storeAddress: storeProfile.address,
    storeDistrict: storeProfile.district,
    storeCity: storeProfile.city,
    storePhone: storeProfile.phone,
    riCompanyName: storeProfile.redInvoice.name,
    riCompanyAddress: storeProfile.redInvoice.address,
    riCompanyDistrict: storeProfile.redInvoice.district,
    riCompanyCity: storeProfile.redInvoice.city,
    riCompanyTaxCode: storeProfile.redInvoice.taxCode
  };
};

function getBase64(img, callback) {
  const reader = new FileReader();
  reader.addEventListener("load", () => callback(reader.result));
  reader.readAsDataURL(img);
}

function EditStoreProfile({
  editStoreProfile,
  storeProfile,
  editStoreProfileStatus,
  onCancel,
  onDone,
  editStoreProfileError
}) {
  const [uploading, setUploading] = useState(false);
  const [imageUrl, setImageUrl] = useState(storeProfile.logoUrl || "");

  const handleChangeImage = info => {
    if (info.file.status === "uploading") {
      setUploading(true);
      return;
    }
    if (info.file.status === "done") {
      getBase64(info.file.originFileObj, imageUrl => {
        setUploading(false);
        setImageUrl(imageUrl);
      });
    }
  };

  if (editStoreProfileStatus === "done") {
    onDone();
  }

  useEffect(() => {
    if (editStoreProfileStatus === "error" && editStoreProfileError) {
      notification.error({
        message: editStoreProfileError.message
      });
    }
  }, [editStoreProfileStatus, editStoreProfileError]);

  const formValues = storeProfile
    ? storeProfileToFormValues(storeProfile)
    : initialFormValues;

  const handleSubmit = values => {
    const data = formValuesToStoreProfile(values);
    data.logoUrl = imageUrl;
    editStoreProfile(data);
  };

  const validateSchema = yup.object().shape({
    storeName: yup.string(),
    storeAddress: yup.string(),
    storeDistrict: yup.string(),
    storeCity: yup.string(),
    storePhone: yup
      .string()
      .matches(
        /^[+]?[(]?[0-9]{3}[)]?[-\s.]?[0-9]{3}[-\s.]?[0-9]{4,6}$/im,
        "Phone Number is invalid"
      ),
    riCompanyName: yup.string(),
    riCompanyAddress: yup.string(),
    riCompanyDistrict: yup.string(),
    riCompanyCity: yup.string(),
    riCompanyTaxCode: yup.string()
  });

  return (
    <div className="pl-16 pr-16 ml-auto mr-auto mt-8" style={{ maxWidth: 900 }}>
      <div>
        <Icon type="edit" className="text-xl text-primary" />
        <span className="ml-4 font-bold">EDIT STORE PROFILE</span>
      </div>
      <Divider />
      <div className="flex justify-between">
        <div className="w-1/3 pr-8">
          <div className="font-bold">STORE IMAGE</div>
          <div className="border rounded overflow-hidden mt-8">
            <img
              alt="store avatar"
              width="100%"
              src={imageUrl || defaultAvatar}
            />
          </div>
          <div className="text-right mt-4">
            <Button type="link" onClick={() => setImageUrl("")}>
              Remove
            </Button>
            <div className="inline-block ml-2">
              <Upload
                action="https://www.mocky.io/v2/5cc8019d300000980a055e76"
                onChange={handleChangeImage}
                showUploadList={false}
              >
                <Button type="primary">
                  {uploading && <Icon type="loading" spin />}Upload
                </Button>
              </Upload>
            </div>
          </div>
        </div>
        <div className="w-2/3 pl-8">
          <Spin spinning={editStoreProfileStatus === "editing"}>
            <div className="font-bold">BASIC INFO</div>
            <div className="mt-8">
              <Formik
                initialValues={formValues}
                onSubmit={handleSubmit}
                validationSchema={validateSchema}
              >
                {({
                  values,
                  handleChange,
                  handleSubmit,
                  setFieldValue,
                  errors
                }) => (
                  <Form onSubmit={handleSubmit} name="editStoreProfileForm">
                    <Form.Item
                      label="Store Name"
                      validateStatus={errors.storeName ? "error" : "success"}
                      help={<ErrorMessage name="storeName" />}
                    >
                      <Input
                        name="storeName"
                        onChange={handleChange}
                        value={values.storeName}
                      />
                    </Form.Item>
                    <Form.Item label="Store Address">
                      <div className="flex justify-start">
                        <div className="w-1/2">
                          <Input
                            placeholder="Address"
                            name="storeAddress"
                            onChange={handleChange}
                            value={values.storeAddress}
                          />
                        </div>
                        <div className="w-1/4 pl-4">
                          <Select
                            placeholder="District"
                            name="storeDistrict"
                            onChange={v => setFieldValue("storeDistrict", v)}
                            value={values.storeDistrict}
                          >
                            <Select.Option value="8">Q.8</Select.Option>
                            <Select.Option value="9">Q.9</Select.Option>
                          </Select>
                        </div>
                        <div className="w-1/4 pl-4">
                          <Select
                            placeholder="City"
                            name="storeCity"
                            onChange={v => setFieldValue("storeCity", v)}
                            value={values.storeCity}
                          >
                            <Select.Option value="HCM">HCM</Select.Option>
                            <Select.Option value="HN">HN</Select.Option>
                          </Select>
                        </div>
                      </div>
                    </Form.Item>
                    <Form.Item
                      label="Phone #"
                      validateStatus={errors.storePhone ? "error" : "success"}
                      help={<ErrorMessage name="storePhone" />}
                    >
                      <Input
                        placeholder="Phone"
                        name="storePhone"
                        onChange={handleChange}
                        value={values.storePhone}
                      />
                    </Form.Item>
                    <div className="mt-8" />
                    <div className="font-bold">RED INVOICE INFO</div>
                    <div className="mt-4" />
                    <Form.Item label="Company Name">
                      <Input
                        name="riCompanyName"
                        onChange={handleChange}
                        value={values.riCompanyName}
                      />
                    </Form.Item>
                    <Form.Item label="Company Address">
                      <div className="flex justify-start">
                        <div className="w-1/2">
                          <Input
                            name="riCompanyAddress"
                            onChange={handleChange}
                            value={values.riCompanyAddress}
                          />
                        </div>
                        <div className="w-1/4 pl-4">
                          <Select
                            placeholder="District"
                            name="riCompanyDistrict"
                            onChange={v =>
                              setFieldValue("riCompanyDistrict", v)
                            }
                            value={values.riCompanyDistrict}
                          >
                            <Select.Option value="8">Q.8</Select.Option>
                            <Select.Option value="9">Q.9</Select.Option>
                          </Select>
                        </div>
                        <div className="w-1/4 pl-4">
                          <Select
                            placeholder="City"
                            name="riCompanyCity"
                            onChange={v => setFieldValue("riCompanyCity", v)}
                            value={values.riCompanyCity}
                          >
                            <Select.Option value="HCM">HCM</Select.Option>
                            <Select.Option value="HN">HN</Select.Option>
                          </Select>
                        </div>
                      </div>
                    </Form.Item>
                    <Form.Item label="MST">
                      <Input
                        name="riCompanyTaxCode"
                        onChange={handleChange}
                        value={values.riCompanyTaxCode}
                      />
                    </Form.Item>
                    <Form.Item>
                      <Button
                        className="w-full"
                        type="primary"
                        htmlType="submit"
                      >
                        Save
                      </Button>
                      <Button className="w-full" type="link" onClick={onCancel}>
                        Cancel
                      </Button>
                    </Form.Item>
                  </Form>
                )}
              </Formik>
            </div>
          </Spin>
        </div>
      </div>
    </div>
  );
}

export default EditStoreProfile;
