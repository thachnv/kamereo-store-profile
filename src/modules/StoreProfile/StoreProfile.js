import React, { useEffect, useState } from "react";
import { compose } from "recompose";
import { connect } from "react-redux";
import "antd/lib/icon/style/css";

import defaultAvatar from "../../assets/default-avatar.png";
import { Button, Modal, Spin } from "antd";
import {
  editStoreProfile,
  getStoreProfile,
  resetEditStoreProfileStatus
} from "./actions";
import WrappedEditStoreProfile from "./EditStoreProfile";

const storeInfoAttributes = [
  {
    label: "Name",
    key: "name",
    id: "store-name"
  },
  {
    label: "Address",
    key: "address",
    id: "store-address"
  },
  {
    label: "Phone #",
    key: "phone",
    id: "store-phone"
  }
];

const redInvoiceAttributes = [
  {
    label: "Name",
    key: "name",
    id: "company-name"
  },
  {
    label: "Address",
    key: "address",
    id: "company-address"
  },
  {
    label: "Tax Code",
    key: "taxCode",
    id: "company-tax-code"
  }
];

export function StoreProfile({
  storeProfile,
  getStoreProfile,
  editStoreProfile,
  editStoreProfileStatus,
  resetEditStoreProfileStatus,
  editStoreProfileError
}) {
  const [showEditForm, setShowEditForm] = useState(false);

  useEffect(() => {
    getStoreProfile();
  }, [getStoreProfile, storeProfile]);

  if (!storeProfile) {
    return (
      <div
        className="p-4 border rounded ml-auto mr-auto flex justify-center items-center"
        style={{ maxWidth: 400, height: 300 }}
      >
        <Spin />
      </div>
    );
  }

  return (
    <div
      className="p-4 border rounded ml-auto mr-auto"
      style={{ maxWidth: 400 }}
    >
      <div className="flex justify-center bg-black">
        <img
          alt="avatar"
          width="100%"
          src={storeProfile.logoUrl || defaultAvatar}
        />
      </div>
      <div className="mt-8 font-bold">STORE INFO</div>
      {storeInfoAttributes.map(attr => (
        <div className="flex justify-start mt-2" key={attr.key}>
          <div className="opacity-50" style={{ width: 200 }}>
            {attr.label}
          </div>
          <div id={attr.id}>{storeProfile[attr.key]}</div>
        </div>
      ))}
      <div className="mt-8 font-bold">RED INVOICE INFO</div>
      {redInvoiceAttributes.map(attr => (
        <div className="flex justify-start mt-2" key={attr.key}>
          <div className="opacity-50" style={{ width: 200 }}>
            {attr.label}
          </div>
          <div id={attr.id}>{storeProfile.redInvoice[attr.key]}</div>
        </div>
      ))}
      <div>
        <Button
          className="w-full mt-8"
          type="default"
          onClick={() => setShowEditForm(true)}
        >
          Edit Profile
        </Button>
      </div>
      <Modal visible={showEditForm} width={900} footer={null} closable={false}>
        <WrappedEditStoreProfile
          editStoreProfileStatus={editStoreProfileStatus}
          editStoreProfile={editStoreProfile}
          storeProfile={storeProfile}
          editStoreProfileError={editStoreProfileError}
          onCancel={() => {
            resetEditStoreProfileStatus();
            setShowEditForm(false);
          }}
          onDone={() => {
            resetEditStoreProfileStatus();
            setShowEditForm(false);
          }}
        />
      </Modal>
    </div>
  );
}
const mapStateToProps = state => ({
  storeProfile: state.storeProfile.storeProfile,
  editStoreProfileStatus: state.storeProfile.editStoreProfileStatus,
  editStoreProfileError: state.storeProfile.editStoreProfileError
});

const mapDispatchToProps = dispatch => ({
  getStoreProfile: () => dispatch(getStoreProfile()),
  resetEditStoreProfileStatus: () => dispatch(resetEditStoreProfileStatus()),
  editStoreProfile: storeProfile => dispatch(editStoreProfile(storeProfile))
});

const WrappedStoreProfile = compose(
  connect(mapStateToProps, mapDispatchToProps)
)(StoreProfile);

export default WrappedStoreProfile;
