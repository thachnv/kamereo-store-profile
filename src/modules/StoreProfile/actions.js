export const getStoreProfile = () => ({
  type: "GET_STORE_PROFILE"
});

export const resetEditStoreProfileStatus = () => ({
  type: "RESET_EDIT_STORE_PROFILE_STATUS"
});

export const editStoreProfile = storeProfile => ({
  type: "EDIT_STORE_PROFILE",
  payload: storeProfile
});
