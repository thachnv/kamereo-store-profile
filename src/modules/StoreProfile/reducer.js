const initialState = {
  storeProfile: null,
  editStoreProfileStatus: "idle"
};

const storeProfile = (state = initialState, action) => {
  switch (action.type) {
    case "GET_STORE_PROFILE_SUCCESS":
      return {
        ...state,
        storeProfile: action.payload
      };
    case "EDIT_STORE_PROFILE":
      return {
        ...state,
        editStoreProfileStatus: "editing"
      };
    case "EDIT_STORE_PROFILE_SUCCESS":
      return {
        ...state,
        storeProfile: action.payload,
        editStoreProfileStatus: "done"
      };
    case "EDIT_STORE_PROFILE_FAILED":
      return {
        ...state,
        editStoreProfileError: action.editStoreProfileError,
        editStoreProfileStatus: "error"
      };
    case "RESET_EDIT_STORE_PROFILE_STATUS":
      return {
        ...state,
        editStoreProfileStatus: "idle"
      };
    default:
      return state;
  }
};

export default storeProfile;
