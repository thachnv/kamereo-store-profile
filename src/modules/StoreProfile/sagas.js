import { call, put, takeEvery } from "redux-saga/effects";
import { fetchStoreProfile, updateStoreProfile } from "../../api/api";

function* getStoreProfile() {
  try {
    const storeProfile = yield call(fetchStoreProfile);
    yield put({
      type: "GET_STORE_PROFILE_SUCCESS",
      payload: storeProfile
    });
  } catch (e) {
    yield put({ type: "GET_STORE_PROFILE_FAILED" });
  }
}
function* editStoreProfile(action) {
  try {
    const storeProfile = yield call(updateStoreProfile, action.payload);
    yield put({
      type: "EDIT_STORE_PROFILE_SUCCESS",
      payload: storeProfile
    });
  } catch (e) {
    yield put({
      type: "EDIT_STORE_PROFILE_FAILED",
      editStoreProfileError: e
    });
  }
}

function* storeProfileSaga() {
  yield takeEvery("GET_STORE_PROFILE", getStoreProfile);
  yield takeEvery("EDIT_STORE_PROFILE", editStoreProfile);
}

export default storeProfileSaga;
