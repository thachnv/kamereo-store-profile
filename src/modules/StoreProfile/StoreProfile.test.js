import React from "react";
import { render } from "@testing-library/react";
import { StoreProfile } from "./StoreProfile";

const mockStoreProfile = {
  id: 1,
  logoUrl: "",
  name: "KOI",
  address: "521 Ho Tung Mau",
  district: "8",
  city: "HCM",
  phone: "1234567890",
  redInvoice: {
    name: "KOI",
    address: "123",
    district: "9",
    city: "HCM",
    taxCode: "123"
  }
};
describe("Test Store Profile view", () => {
  it("should render without failure", () => {
    const rendered = render(
      <StoreProfile
        storeProfile={mockStoreProfile}
        getStoreProfile={jest.fn()}
      />
    );
    expect(rendered).toMatchSnapshot();
  });

  it("should call method to get data only once", async () => {
    const mockGetStoreProfile = jest.fn();
    render(
      <StoreProfile
        storeProfile={mockStoreProfile}
        getStoreProfile={mockGetStoreProfile}
      />
    );
    // await waitForElement(() => getByText(rendered.container, "STORE INFO"));
    expect(mockGetStoreProfile).toBeCalledTimes(1);
  });

  it("should fill data correctly", () => {
    const rendered = render(
      <StoreProfile
        storeProfile={mockStoreProfile}
        getStoreProfile={jest.fn()}
      />
    );
    expect(rendered.container.querySelector("#store-name").innerHTML).toBe(
      mockStoreProfile.name
    );
    expect(rendered.container.querySelector("#store-address").innerHTML).toBe(
      mockStoreProfile.address
    );
    expect(rendered.container.querySelector("#store-phone").innerHTML).toBe(
      mockStoreProfile.phone
    );
    expect(rendered.container.querySelector("#company-name").innerHTML).toBe(
      mockStoreProfile.redInvoice.name
    );
    expect(rendered.container.querySelector("#company-address").innerHTML).toBe(
      mockStoreProfile.redInvoice.address
    );
    expect(
      rendered.container.querySelector("#company-tax-code").innerHTML
    ).toBe(mockStoreProfile.redInvoice.taxCode);
  });
});
