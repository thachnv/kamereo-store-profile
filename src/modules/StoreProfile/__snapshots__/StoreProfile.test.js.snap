// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Store Profile view should render without failure 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="p-4 border rounded ml-auto mr-auto"
        style="max-width: 400px;"
      >
        <div
          class="flex justify-center bg-black"
        >
          <img
            alt="avatar"
            src="default-avatar.png"
            width="100%"
          />
        </div>
        <div
          class="mt-8 font-bold"
        >
          STORE INFO
        </div>
        <div
          class="flex justify-start mt-2"
        >
          <div
            class="opacity-50"
            style="width: 200px;"
          >
            Name
          </div>
          <div
            id="store-name"
          >
            KOI
          </div>
        </div>
        <div
          class="flex justify-start mt-2"
        >
          <div
            class="opacity-50"
            style="width: 200px;"
          >
            Address
          </div>
          <div
            id="store-address"
          >
            521 Ho Tung Mau
          </div>
        </div>
        <div
          class="flex justify-start mt-2"
        >
          <div
            class="opacity-50"
            style="width: 200px;"
          >
            Phone #
          </div>
          <div
            id="store-phone"
          >
            1234567890
          </div>
        </div>
        <div
          class="mt-8 font-bold"
        >
          RED INVOICE INFO
        </div>
        <div
          class="flex justify-start mt-2"
        >
          <div
            class="opacity-50"
            style="width: 200px;"
          >
            Name
          </div>
          <div
            id="company-name"
          >
            KOI
          </div>
        </div>
        <div
          class="flex justify-start mt-2"
        >
          <div
            class="opacity-50"
            style="width: 200px;"
          >
            Address
          </div>
          <div
            id="company-address"
          >
            123
          </div>
        </div>
        <div
          class="flex justify-start mt-2"
        >
          <div
            class="opacity-50"
            style="width: 200px;"
          >
            Tax Code
          </div>
          <div
            id="company-tax-code"
          >
            123
          </div>
        </div>
        <div>
          <button
            class="ant-btn w-full mt-8 ant-btn-default"
            type="button"
          >
            <span>
              Edit Profile
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="p-4 border rounded ml-auto mr-auto"
      style="max-width: 400px;"
    >
      <div
        class="flex justify-center bg-black"
      >
        <img
          alt="avatar"
          src="default-avatar.png"
          width="100%"
        />
      </div>
      <div
        class="mt-8 font-bold"
      >
        STORE INFO
      </div>
      <div
        class="flex justify-start mt-2"
      >
        <div
          class="opacity-50"
          style="width: 200px;"
        >
          Name
        </div>
        <div
          id="store-name"
        >
          KOI
        </div>
      </div>
      <div
        class="flex justify-start mt-2"
      >
        <div
          class="opacity-50"
          style="width: 200px;"
        >
          Address
        </div>
        <div
          id="store-address"
        >
          521 Ho Tung Mau
        </div>
      </div>
      <div
        class="flex justify-start mt-2"
      >
        <div
          class="opacity-50"
          style="width: 200px;"
        >
          Phone #
        </div>
        <div
          id="store-phone"
        >
          1234567890
        </div>
      </div>
      <div
        class="mt-8 font-bold"
      >
        RED INVOICE INFO
      </div>
      <div
        class="flex justify-start mt-2"
      >
        <div
          class="opacity-50"
          style="width: 200px;"
        >
          Name
        </div>
        <div
          id="company-name"
        >
          KOI
        </div>
      </div>
      <div
        class="flex justify-start mt-2"
      >
        <div
          class="opacity-50"
          style="width: 200px;"
        >
          Address
        </div>
        <div
          id="company-address"
        >
          123
        </div>
      </div>
      <div
        class="flex justify-start mt-2"
      >
        <div
          class="opacity-50"
          style="width: 200px;"
        >
          Tax Code
        </div>
        <div
          id="company-tax-code"
        >
          123
        </div>
      </div>
      <div>
        <button
          class="ant-btn w-full mt-8 ant-btn-default"
          type="button"
        >
          <span>
            Edit Profile
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
